controllers/cluster_controller.go

```diff
@@ -238,8 +238,16 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 
 			gvk := child.GetObjectKind().GroupVersionKind().String()
 
-			logger.Info("Deleting child", "gvk", gvk, "name", accessor.GetName())
-			if err := r.Client.Delete(context.Background(), child); err != nil {
+			childObject, ok := child.(client.Object)
```

Comment:
Do you need this type assertion? Isn't child already a client.Object?

Also, we shouldn't need to use meta.Accessor any more?

Reply:
`child` is a `metav1.Object` in this case

Reply:
I see what you mean, yeah let me change that